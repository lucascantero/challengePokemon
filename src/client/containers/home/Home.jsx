import React, { useEffect, useState } from 'react';
import Card from '../../components/card/Card'
import { useTranslation } from 'react-i18next';
import Header from '../../components/header/Header'

const Home = () => {

    const [data, setData] = useState([]);
    const [offset, setOffset] = useState(0);
    const { t } = useTranslation();

    useEffect(() => {
        getData();
    }, [offset]);

    const getData = async () => {

        const response = await fetch(`/api/pokemon?offset=${offset}`);
        const dataAPI = await response.json();
        setData(dataAPI);
    }

    return (
        <>
            <Header />
            <div className="home__contenedor">
                <div className="home__wrapper">
                    <ul className="home__ul">
                        {
                            data.map(element => (
                                <Card key={element.name} dato={element} />
                            ))
                        }
                    </ul>
                    <div className="home__groupButton">
                        <button className="home__button" disabled={!offset} onClick={() => setOffset(offset - 5)}>
                            {t('previous')}
                        </button>
                        <button className="home__button" onClick={() => setOffset(offset + 5)}>
                            {t('next')}
                        </button>
                    </div>
                </div>
            </div>
        </>
    )
}

export default Home;