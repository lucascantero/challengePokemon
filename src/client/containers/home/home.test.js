import React from 'react';
import Home from './Home.jsx';
import { act } from 'react-dom/test-utils';
import { render, unmountComponentAtNode } from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import { I18nextProvider } from 'react-i18next';
import i18next from '../../i18n/index';


describe('Testing Home', () => {

    let container = null;

    beforeEach(() => {
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });


    it("Should match snapshot", async () => {

        const fakeUser = [{
            "name": "name",
            "id": "id",
            "img": "img"
        }]

        global.fetch = jest.fn().mockImplementation
            (() =>
                Promise.resolve({
                    json: (() => Promise.resolve(fakeUser))
                })
            );

        await act(async () => {
            return render(<MemoryRouter> <I18nextProvider i18n={i18next}> <Home /> </I18nextProvider></MemoryRouter>, container);
        });

        expect(container.innerHTML).toMatchSnapshot();
    })
})