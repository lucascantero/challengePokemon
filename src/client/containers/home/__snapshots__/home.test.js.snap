// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home Should match snapshot 1`] = `"  <div class=\\"searchBox__contenedor\\"><div class=\\"buttonLanguage__wrapper\\"><button class=\\"buttonLanguage__button\\">Inglés</button><button class=\\"buttonLanguage__button\\">Español</button></div></div><div class=\\"home__contenedor\\"><div class=\\"home__wrapper\\"><ul class=\\"home__ul\\"><li class=\\"card__li\\"><a class=\\"card__link\\" href=\\"/pokemon/id\\"><div class=\\"card__wrapper\\"><img src=\\"img\\" alt=\\"pokemon\\" class=\\"card__img\\"><h1 class=\\"card__name\\">name</h1></div></a></li></ul><div class=\\"home__groupButton\\"><button class=\\"home__button\\" disabled=\\"\\">Anterior</button><button class=\\"home__button\\">Siguiente</button></div></div></div> "`;
